src/extensionProvider.ts

```diff
@@ -63,7 +61,7 @@ export class ExtensionProvider {
     const { token } = options;
     const data = token ? token : "";
     await this.startBgrMsgChannel(Operation.Connect, data);
-    return this.account.address;
+    return this.account;
```

Comment:
what about signature?

Reply:
It's already handled. Please check line:201
`this.account = event.data.data;`